packages/terra-i18n-plugin/src/I18nAggregatorPlugin.js

```diff
@@ -0,0 +1,59 @@
+import fs from 'fs';
+import path from 'path';
+import mkdirp from 'mkdirp';
+
+import supportedLocales from 'terra-i18n/lib/i18nSupportedLocales';
+
+function generateTranslationFile(language, messages) {
+  return `import { addLocaleData } from 'react-intl';
+import localeData from 'react-intl/locale-data/${language.split('-')[0]}';
+
+addLocaleData(localeData);
+
+const messages = ${JSON.stringify(messages, null, 2)};
+
+module.exports = {
+  areTranslationsLoaded: true,
+  locale: '${language}',
+  messages,
+};`;
+}
+
+function getDirectories(srcPath) {
+  return fs.readdirSync(srcPath).filter(file => fs.statSync(path.join(srcPath, file)).isDirectory());
+}
+
+function apply(options) {
+  supportedLocales.forEach((language) => {
+    const currentLanguageMessages = {};
+    getDirectories(path.resolve(options.baseDirectory, 'node_modules')).forEach((module) => {
+      const translationFile = path.resolve(options.baseDirectory, 'node_modules', module, 'translations', `${language}.json`);
+      if (fs.existsSync(translationFile)) {
+        Object.assign(currentLanguageMessages, JSON.parse(fs.readFileSync(translationFile, 'utf8')));
+      }
+    });
+
+    if (currentLanguageMessages !== {}) {
```

Comment:
I know it may seem weird, but I think it's still worth generating the file for a supported translation even if the messages are empty. Some applications have a hard time dealing with determining how to handle the case when a requested file is not found. Anyone else have thoughts on it?